app/assets/javascripts/newflow/index.js

```diff
@@ -36,6 +36,26 @@ $(document).ready(function(){
     }
   });
 
+  // Validate name for unallowed chars
+  var errorDiv = '<div class="errors invalid-message">Name cannot contain special characters</div>';
+  function check_for_specialness(obj) {
+      if(/^[a-zA-Z0-9- ]*$/.test(obj.value) == false) {
+          $(obj).addClass('has-error');
+          $(obj).closest('.control-group').append(errorDiv);
+      }else{
+          $(obj).removeClass('has-error');
+          $(obj).closest('.control-group').remove(errorDiv);
+      }
+  }
+  $("#signup_first_name").blur(function (e) {
```

Comment:
yeah - the `else` statement there is removing the error div :) 
I think onChange would cause it to update too frequently 
might want to checkout the branch and check how it's working

Reply:
That's not really a problem — onChange is a very common thing to do that doesn't cause that cost that much in performance. See what we're doing for step 4: https://github.com/openstax/accounts/blob/master/app/assets/javascripts/newflow/educator_complete_dynamic.js.coffee

Reply:
I can see what the else statement is doing. In other words, what I meant is, could you do what the else statement is doing but onChange instead on blur?